default.nix

```diff
@@ -0,0 +1,4 @@
+{ pkgs ? import <nixpkgs> {}, writeShellScriptBin ? pkgs.writeShellScriptBin, maven ? pkgs.maven }:
+writeShellScriptBin "mvn2nix" ''
+     MAVEN_OPTS=-Dorg.slf4j.simpleLogger.logFile=System.err ${maven}/bin/mvn -Dmaven.repo.local=$(mktemp -d -t mavenXXX) org.nixos.mvn2nix:mvn2nix-maven-plugin:mvn2nix "$@"
+''
```

Comment:
This is merely step 1; next round will be to get the plugin to build itself using the _package-info.json_ it has produced.
Then setting the _maven.repo.local_ to that value.

In this world, the version identifier will be frozen to something like 0.0-NIX
(I'll use set-version during the build process to change the version in the nix derivation)